compiler/rustc_passes/src/reachable.rs

```diff
@@ -197,10 +198,20 @@ impl<'tcx> ReachableContext<'tcx> {
                     // Reachable constants will be inlined into other crates
                     // unconditionally, so we need to make sure that their
                     // contents are also reachable.
-                    hir::ItemKind::Const(_, _, init) | hir::ItemKind::Static(_, _, init) => {
+                    hir::ItemKind::Const(_, _, init) => {
                         self.visit_nested_body(init);
                     }
 
+                    // Reachable statics are inlined if read from another constant or static
+                    // in other crates. Additionally anonymous nested statics may be created
+                    // when evaluating a static, so preserve those, too.
```

Comment:
How is it sound to "inline" statics that are defined in a different crate?

Reply:
well... what I mean to say is that when you do

```rust
static FOO:() = {
    STATIC_FROM_OTHER_CRATE;
};
```

you're copying that other crate's static evaluated initializer value right here instead of reevaluating it.

Reply:
Okay. That's the opposite of inlining IMO -- inlining would mean inserting the code (like function inlining).

Even then I don't understand how that's related to what is considered "reachable", or to the code below the comment. I can't review this part anyway since it's related to monomorphization, but it still wouldn't hurt if I could at least make sense of the comment. ;)

Why do we even have to visit the body if only the final value matters?

Reply:
We're calling `visit_nested_body`. If that doesn't visit the body then the function might need a rename.^^